Add missing references for support of FanSystemModel in various HVAC/ZoneHVAC components

jmarrec opened the pull request:
Pull request overview
---------------------
 - Fixes #7697
 - Add missing references for support of FanSystemModel in various components
 - Also propagated upstream some differences we had had to make in OpenStudio/resources/energyplus/ProposedEnergy+.idd to avoid failures in our forward translator (which is more strictly enforcing IDD references than E+)
 - For context, the needs for these changes have been discovered while writting up the third (and hopefully) PR to support FanSystemModel in OpenStudio: https://github.com/NREL/OpenStudio/pull/3864. I also wrote a regression test in parallel in https://github.com/NREL/OpenStudio-resources/pull/92 that should test all (I think) 17 HVAC/ZoneHVACComponent that support it with a FanSystemModel

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

jmarrec committed:
Propagate Changes we had to do in OpenStudio (as of 2.9.1) to make things work: missing plant equipment references etc

Note that for solar collectors, technically E+ does say it expects them on the demand side, but it works on the supply...

https://github.com/NREL/OpenStudio/pull/3762
https://github.com/NREL/OpenStudio/issues/3761#issuecomment-553814770
https://github.com/NREL/OpenStudio/issues/2880

jmarrec committed:
Caught another missing ref

jmarrec committed:
PlantEquipmentOperation:XXX should accept both CondenserEquipmentList and PlantEquipmentList nowadays

jmarrec committed:
AirflowNetwork:Distribution:Component:Fan doesn't seem to support Fan:COmponentModel, so update ref to more restrictive than "Fans"

jmarrec committed:
Correct IDD references to FanSystemModel

jmarrec committed:
ZoneHVAC:UnitVentilator and UnitHeater missing ref to FanOnOff which is accepted

jmarrec committed:
ZoneHVAC:OutdoorAir unit exhaust fan actually accepts FanSystemModel as well( checked OutdorAirUnit.cc)

jmarrec committed:
FaultModel:Fouling:AirFilter accepts FanSystemModel

cf https://github.com/jmarrec/EnergyPlus/blob/6d979341f3d85973fca8f1e7afc02a6484be4bb4/src/EnergyPlus/HVACFan.cc#L732

jmarrec committed:
FanPerformance:NightVentilation seems to be used by Fan:CV Fan:VV and Fan:COmponentModel actually...

jmarrec committed:
Merge branch 'develop' into 7697_FanSystemModel_IDD

jmarrec committed:
Strip trailing whitespace

jmarrec committed:
Rescan changes once more

jmarrec committed:
Merge remote-tracking branch 'upstream/develop' into 7697_FanSystemModel_IDD

jmarrec committed:
Merge remote-tracking branch 'upstream/develop' into 7697_FanSystemModel_IDD

jmarrec committed:
Merge remote-tracking branch 'upstream/develop' into 7697_FanSystemModel_IDD

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -72534,7 +72568,7 @@ CoolingTower:VariableSpeed:Merkel,
        \note This field is only used if the previous is set to autocalculate and performance input method is NominalCapacity
        \units m3/s-W
-       \ip-units (gal/min)/(Btu/h)
+       \ip-units gal/min-W
```
No, I'm fairly certain this is intentionally `(gal/min)/(Btu/h)`.  Because the nominal capacity switches to Btu/h in ip-units.

jmarrec replied:
That's a good point! I just copied over an existing diff we had in our ProposedEnergy+.idd. I didn't think twice... I suppose MEPs would use GPM/ton in the US for this.

jmarrec replied:
Reverted in 36bd6bd89592856de2031fe9d92a1afa398dec7d

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -85331,6 +85369,10 @@ Generator:MicroTurbine,
+      \reference-class-name validPlantEquipmentTypes
+      \reference validPlantEquipmentNames
+      \reference-class-name validCondenserEquipmentTypes
+      \reference validCondenserEquipmentNames
```
I don't think this is correct. Generators with waste heat recovery can sit on a branch, but they are not dispatchable by the load management. So, I don't think they belong in plant or condenser equipment lists. @Myoldmopar ?

jmarrec replied:
If you use an `ElectricLoadCenter:Distribution` object with a scheme like `FollowThermalLimitElectric`, then you do have to put in a `PlantEquipmentOperation:HeatingLoad` don't you?

That's what I'm doing in the OpenStudio-resources regression test for Generator:MicroTurbine at least... https://github.com/NREL/OpenStudio-resources/blob/9d16ac6aba234794c84438a679ce4ff54710dcc5/model/simulationtests/generator_microturbine.rb#L385

mjwitte replied:
I'm not really sure if that does anything. You may be correct. @Myoldmopar @EnergyArchmage ?

MatthewSteen replied:
I happen to be working with the Generator:MicroTurbine by adding it to the MediumOffice reference building (using Julien's `generator_microturbine.rb` code, thanks btw!), and can confirm that the `PlantEquipmentOperation:HeatingLoad` is needed for the `ElectricLoadCenter:Distribution` > `FollowThermalLimitElectrical` case to have _any_ effect on a model.

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -88925,11 +88967,13 @@ FaultModel:Fouling:AirFilter,
        \key Fan:OnOff
        \key Fan:ConstantVolume
        \key Fan:VariableVolume
+       \key Fan:SystemModel
```
I don't think this is correct. See https://github.com/NREL/EnergyPlus/issues/7697#issuecomment-578222135

jmarrec replied:
I jumped the gun indeed, made it a `\note` instead in https://github.com/NREL/EnergyPlus/commit/36bd6bd89592856de2031fe9d92a1afa398dec7d#diff-23ccf090b80d26e885712256b9a6d888R88970

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -68354,6 +68382,8 @@ SolarCollector:FlatPlate:Water,
        \reference-class-name validBranchEquipmentTypes
        \reference validBranchEquipmentNames
+       \reference-class-name validPlantEquipmentTypes
+       \reference validPlantEquipmentNames
```
Not sure solar collectors belong on PlantEquipmentLists? @mitchute @Myoldmopar ?

mitchute replied:
Looks to me like it *should* be there: https://github.com/NREL/EnergyPlus/blob/bcfdd258b7b37bd9bb547835c3c9bb4cd9367ad1/src/EnergyPlus/SolarCollectors.hh#L128

jmarrec replied:
Lots of confusion on what the placement of solar collectors should be, especially since OpenStudio forces you to put the Flat Plate ones on the supply side (for some reason... and it appears to work fine too...)
I agree that generally speaking the documentation and example files in E+ seem to imply demand side placement (with control of the loop via availability managers).

References:

* https://unmethours.com/question/27837/which-side-of-the-plant-loop-for-solar-collector-in-eopenstudio/
* NREL/OpenStudio#2880
* https://github.com/NREL/OpenStudio/issues/3761#issuecomment-553666339

jmarrec replied:
Opening a dedicated issue for this instead: https://github.com/NREL/EnergyPlus/issues/7735

mjwitte reviewed (commented):
@jmarrec Not sure if this was ready for review yet, but made some comments anyway.

jmarrec committed:
Merge remote-tracking branch 'upstream/develop' into 7697_FanSystemModel_IDD

jmarrec committed:
Address a couple of review comments from @mjwitte

EnergyArchmage commented:
It is possible for a chp generator to be operated in a thermal load following scheme.  I think @jmarrec is correct. Just glanced at comments.

jmarrec commented:
in `Microgeneration.idf`, the `Generator:MicroCHP` for eg is on a `PlantEquipmentList` used in a `PlantEquipmentOperation:HeatingLoad`.

(Note that I added the reference for Generator:MicroTurbine but not for Generator:MicroCHP simply because I modified that stuff while testing things that are implemented in OS, and MicroTurbine is but MicroCHP isn't. **Edit**: actually MicroCHP already has the same references)

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 7697_FanSystemModel_IDD

mjwitte reviewed (approved):
@jmarrec I'm ok with the responses to prior comments. I've tested this with IDF Editor validity check for about 2 dozen idf files that contain Fan:SystemModel. All passed without errors. I've updated to current develop and will wait for one more round of CI results, then merge this.
